eng/pipelines/templates/steps/generate-doc.yml

```diff
@@ -15,7 +15,17 @@ steps:
   - pwsh: |
       node $(Build.SourcesDirectory)/eng/tools/generate-doc/dist/index.js --serviceDir "${{parameters.ServiceDirectory}}"
     displayName: "Run Typedoc Docs"
+    condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+
+  - pwsh: |
+      if ("$(ChangedServices)") {
+        foreach($service in "$(ChangedServices)".Split(" ")) {
```

Comment:
```suggestion
        foreach($service in '$(ChangedServices)'.Split(" ")) {
```

Reply:
Related to this and my above comment, it might be better to just do it like:

```
if ($env:ChangedServices) {}
...
env:
  ChangedServices: $(ChangedServices)
```